tests/unit/components/DocumentationTopic/DocumentationNav/VersionToggle.spec.js

```diff
@@ -0,0 +1,215 @@
+  let wrapper;
+
+  const propsData = {
+    versionList: ['V3, V2, V1'],
+  };
+
+  const mocks = {
+    $route: {
+      path: '/documentation/foo',
+    },
+    $router: {
+      push: jest.fn(),
+    },
+  };
+
+  const provide = {
+    store: {
+      setPreferredVersion: jest.fn(),
+    },
+  };
+
+  const stubs = {
+    'router-link': RouterLinkStub,
+  };
+
+  const createWrapper = (props = propsData, mocksData = mocks) => (
+    shallowMount(VersionToggle, {
+      mocks: mocksData,
+      propsData: props,
+      provide,
+      stubs,
+    })
+  );
+
+  beforeEach(() => {
+    wrapper = createWrapper();
+    localStorage.clear();
+    DocumentationTopicStore.reset();
+  });
+
+  it('renders `NavMenuItemBase` at the root', () => {
+    const menuItemBase = wrapper.find(NavMenuItemBase);
+    expect(menuItemBase.exists()).toBe(true);
+    expect(menuItemBase.classes()).toContain('nav-menu-setting');
+    expect(menuItemBase.classes()).toContain('version-container');
+  });
+
+  it('does not render the version toggle component when there are no versions', () => {
+    const toggle = wrapper.find('#version-toggle');
+    expect(toggle.exists()).toBe(false);
+  });
+
+  it('renders the single version element when the page contains single page', () => {
+    wrapper.setProps({
+      versionList: ['V3'],
+    });
+    const toggle = wrapper.find('#single-version-page');
+    expect(toggle.exists()).toBe(true);
+  });
+
+  it('renders the version toggle element when the page contains multiples pages', async () => {
+    wrapper.setProps({
+      versionList: ['V3', 'V2'],
+    });
+    await flushPromises();
+    const toggle = wrapper.find('#version-toggle');
+    expect(toggle.exists()).toBe(true);
+  });
+
+  it('renders options inside select for each version', () => {
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+    const options = wrapper.findAll('#version-toggle option');
+    expect(options.length).toBe(3);
+
+    expect(options.at(0).attributes('value')).toBe('V3');
+    expect(options.at(0).text()).toBe('V3');
+
+    expect(options.at(1).attributes('value')).toBe('V2');
+    expect(options.at(1).text()).toBe('V2');
+
+    expect(options.at(2).attributes('value')).toBe('V1');
+    expect(options.at(2).text()).toBe('V1');
+  });
+
+  it('calls router and changes v-model when different option is selected', async () => {
+    wrapper.setProps({
+      versionList: ['V3', 'V2', 'V1'],
+    });
+
+    expect(wrapper.find('#version-toggle').element.value).toBe('V3');
+
+    wrapper.findAll('#version-toggle option').at(2).setSelected();
+
+    const options = wrapper.find('#version-toggle').findAll('option');
+
+    await options.at(2).setSelected();
+
+    expect(wrapper.find('#version-toggle').element.value).toBe('V1');
+    expect(mocks.$router.push).toHaveBeenCalledWith({ path: '/documentation/foo', query: { version: 'V1' } });
+    expect(provide.store.setPreferredVersion).toHaveBeenCalledWith('V1');
```

Comment:
OK so you are finding the item at(2), which is the third item, in your case `V1`. You dont need to call `setSelected` twice, just await the one of them.

```suggestion
    const versionToggle = wrapper.find('#version-toggle');
    const options = versionToggle.findAll('option');

    await options.at(2).setSelected();

    expect(versionToggle.element.value).toBe('V1');
    expect(mocks.$router.push).toHaveBeenCalledWith({ path: '/documentation/foo', query: { version: 'V1' } });
    expect(provide.store.setPreferredVersion).toHaveBeenCalledWith('V1');
```